python/cudf/cudf/_lib/column.pxd

```diff
@@ -14,7 +14,18 @@ from cudf._lib.cpp.column.column_view cimport (
 
 
 cdef class Column:
-    cdef dict __dict__
+    cdef public:
+        cdef int _offset
+        cdef int _size
+        cdef object _dtype
+        cdef object _base_children
+        cdef object _base_data
+        cdef object _base_mask
+        cdef object _children
+        cdef object _data
+        cdef object _mask
+        cdef object _null_count
```

Comment:
We should do the same for `table.pyx`

Reply:
So what makes this difficult/impossible right now is `RangeIndex`, which redefines `_data` as a property.

When `cdef` methods try to access the `._data` attributes, they don't evaluate the property, but rather the `cdef` attribute which breaks everything.